pt_miniscreen/pages/root/projects.py

```diff
@@ -0,0 +1,178 @@
+                exit_condition=project_config.get("exit_condition"),
+            )
+        except Exception as e:
+            logger.warning(f"Error parsing file '{file}': {e}")
+            raise InvalidConfigFile(e)
+
+
+class ProjectPage(Component):
+    def __init__(self, project_config: ProjectConfig, **kwargs):
+        self.project_config = project_config
+        super().__init__(**kwargs)
+        self.text = self.create_child(
+            MarqueeText,
+            text=self.project_config.title,
+            font_size=14,
+            align="center",
+            vertical_align="center",
+        )
+        self.image = self.create_child(
+            Image,
+            image_path=self.project_config.image,
+        )
+
+    def render(self, image):
+        return apply_layers(
+            image,
+            [
+                layer(
+                    self.text.render,
+                    size=(120, 25),
+                    pos=(28, 0),
+                ),
+                layer(
+                    self.image.render,
+                    size=(25, 25),
+                    pos=(0, 0),
+                ),
+            ],
+        )
+
+
+class ProjectRow(Component):
+    def __init__(self, project_config: ProjectConfig, **kwargs) -> None:
+        self.project_config = project_config
+        super().__init__(**kwargs)
+        self.text = self.create_child(
+            MarqueeText,
+            text=self.project_config.title,
+            font_size=10,
+            align="center",
+            vertical_align="center",
+        )
+        self.page = partial(ProjectPage, self.project_config)
+
+    def render(self, image):
+        return apply_layers(
+            image,
+            [
+                layer(
+                    self.text.render,
+                    size=(128, 10),
+                    pos=(0, 0),
+                ),
+            ],
+        )
+
+
+class EmptyProjectRow(Component):
+    def __init__(self, **kwargs) -> None:
+        super().__init__(**kwargs)
+        self.page = None
+        self.text = self.create_child(
+            MarqueeText,
+            text="No projects found",
+            font_size=10,
+            align="center",
+            vertical_align="center",
+        )
+
+    def render(self, image):
+        return apply_layers(
+            image,
+            [
+                layer(
+                    self.text.render,
+                    size=(128, 10),
+                    pos=(0, 0),
+                ),
+            ],
+        )
+
+
+class ProjectList(SelectableList):
+    PROJECT_DIRECTORIES = [
+        "/home/pi/Desktop/Projects/",
+    ]
+
+    def __init__(self, **kwargs) -> None:
+        super().__init__(
+            Rows=self.load_project_rows(),
+            num_visible_rows=5,
+            **kwargs,
+        )
+
+    def load_project_rows(self) -> List:
+        pages: List[Union[Type[EmptyProjectRow], partial[ProjectRow]]] = []
```

Comment:
I think this variable should be called rows rather than pages (picky of me sorry)